examples/resolver/resolver.go

```diff
@@ -0,0 +1,28 @@
+package resolver
+
+import (
+	"fmt"
+	"net/http"
+)
+
+//RequestURL simply returns
+//the request url from REQUEST_URI header
+//this should not be done in production applications
+type RequestURL struct {
```

Comment:
But it would be a little more complex than that. It should have a whitelist of allowed TLDs and maybe some regex handling .. don't you think?